src/tarantool-client/Converters/EnumConverter.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Globalization;
+
+using TarantoolDnx.MsgPack;
+
+namespace tarantool_client.Converters
+{
+    public class EnumConverter<T> : IMsgPackConverter<T>
+        where T : struct, IConvertible
+    {
+        public void Write(T value, IBytesWriter writer, MsgPackContext context)
```

Comment:
Checking for IsEnum added.
Enum underlaying type extracted and used in serialization/deserialization process.